server/api_webex.go

```diff
@@ -0,0 +1,171 @@
+	if sessionUserID == "" {
+		JSONErrorResponse(w, ErrNotAuthorized, http.StatusUnauthorized)
+		return
+	}
+
+	var req StartMeetingRequest
+
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		JSONErrorResponse(w, err, http.StatusBadRequest)
+		return
+	}
+
+	_, err := p.loadWebexSession(sessionUserID)
+	switch {
+	case
+		err == ErrWebexSessionNotFound,
+		err == ErrWebexUserNotFound:
+
+		p.API.LogError("Error retrieving session", "err", err.Error())
+		JSONErrorResponse(w, ErrWebexNotConnected, http.StatusUnauthorized)
+		return
+
+	case err != nil:
+		JSONErrorResponse(w, err, http.StatusInternalServerError)
+		return
+
+	}
+
+	user, err := p.loadWebexUser(sessionUserID)
+	switch {
+	case
+		err == ErrWebexSessionNotFound,
+		err == ErrWebexUserNotFound:
+		JSONErrorResponse(w, ErrWebexNotConnected, http.StatusUnauthorized)
+		return
+	case err != nil:
+		JSONErrorResponse(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	channel, appErr := p.API.GetChannel(req.ChannelId)
+	if appErr != nil {
+		JSONErrorResponse(w, appErr, http.StatusInternalServerError) // maybe bad request
+		return
+	}
+
+	// Not sure how reliable this is...
+	toUserID := ""
+	parts := strings.Split(channel.Name, "__")
+	if len(parts) > 0 {
+		toUserID = parts[0]
+		if toUserID == sessionUserID && len(parts) > 1 {
+			toUserID = parts[1]
+		}
+	}
+
+	if toUserID == "" {
+		JSONErrorResponse(w, ErrCreateMeetingToUserIdNotFound, http.StatusBadRequest)
+		return
+	}
+
+	if toUserID == sessionUserID {
+		JSONErrorResponse(w, ErrCreateMeetingFailedOwnChannel, http.StatusBadRequest)
+		return
+	}
+
+	isGuest := false
+
+	toWebexUser, err := p.loadWebexUser(toUserID)
+	switch {
+	case err == ErrWebexUserNotFound:
+		isGuest = true
+	case err != nil:
+		JSONErrorResponse(w, err, http.StatusInternalServerError)
+		return
+	}
+
+	meeting := WebexMeeting{
+		ID:            model.NewId(),
+		ChannelID:     channel.Id,
+		FromUserID:    sessionUserID,
+		FromWebexUser: *user,
+		ToUserID:      toUserID,
+		ToWebexUser:   *toWebexUser,
+	}
+
+	if isGuest {
+		toUser, err := p.API.GetUser(toUserID)
+		if err != nil {
+			JSONErrorResponse(w, err, http.StatusInternalServerError)
+			return
+		}
+		meeting.GuestEmail = toUser.Email
+	}
+
+	siteURL := p.API.GetConfig().ServiceSettings.SiteURL
+
+	meeting.URL = fmt.Sprintf("%s/plugins/%s/meetings/%s", *siteURL, manifest.Id, meeting.ID)
+
+	post := &model.Post{
+		UserId:    sessionUserID,
+		ChannelId: channel.Id,
+		Message:   fmt.Sprintf("Webex Meeting started: %s", meeting.URL),
+		Type:      "custom_zoom",
+		Props: map[string]interface{}{
+			"meeting_id":        meeting.ID,
+			"meeting_link":      meeting.URL,
+			"meeting_status":    "STARTED",
+			"meeting_personal":  false,
+			"meeting_topic":     "Webex Meeting",
+			"from_webhook":      "true",
+			"override_username": "Webex",
+			"override_icon_url": "https://smartsolutioncomputer.com/upload-img/Products/Cisco/Webex-logo.png",
```

Comment:
Can we host this locally? It won't load in some environments.

Reply:
I've updated it to use the static asset from the webapp side.